### DISCUSSION TITLE
Merging paths

### DISCUSSION THREAD

**Discussion by andrewbaxter** [GraphiteEditor/Graphite#2158] [Help] [answered]
So, I realize this is still very early in development, but I figured no matter how many issues there were it'd be better than doing 10 variations of this design manually in inkscape...

I have a bunch of letter strokes.  I want to adjust all their line thicknesses together (it's one word, the style for all the strokes should be the same).  Selecting them as a group hides the stroke properties.  In inkscape or whatever I'd try unioning the shapes, but I can't find a way to do a destructive union, or apply the boolean operator (I assume that works at a different step in the pipeline anyway so not sure it'd work in the first place).

My current workaround was to make a bunch of merge nodes and merge the transformed path data before sending it to a single stroke node, but that took a decent amount of clicking and I'm not sure if there will be other fallout from having multiple path objects.

FWIW I think (dumb) unioning paths should be fairly safe assuming there's a "split disconnected segments" action too.

**Comment by Keavon** [comment, marked as answer]
Thanks for posting so we know where your pain points are! This should improve in the future through the addition of more tooling. Here's how to do it now:

1. Group the text layers and select the group
2. Click the !["Add an operation to the end of this layer's chain of nodes"](https://github.com/user-attachments/assets/3fea54ca-78bf-4a60-a09f-a06e62b3a9f0) button in the top right of the Properties panel
3. Search for "Stroke" and click to add it
4. Set your choice of stroke parameters to apply to each text layer's path within the group

**Reply by andrewbaxter** [reply]
Ah okay, I tried that and it worked!

So does that mean there's no way to join paths into a single object? Also, now there's 2 strokes in line with eachother (path -> transform -> stroke -> layer(/group?) -> stroke - I guess the later stroke overrides the earlier? I'd assume that after the first stroke it's now a path representing the outline of the first stroke, and so I'm outlining the outline.  Maybe I needed to delete the first strokes first?

Also, I didn't see a group option anywhere.  I tried `ctrl+g` and it worked, but I first checked all the buttons in node view + not, right clicking the layers in the layer tree, and looking through all the menus and didn't see anything (and nothing in the hotkeys at the bottom, although they go off my screen - 1920 non 4k display).

**Reply by Keavon** [reply]
> So does that mean there's no way to join paths into a single object?

You can use the *Flatten Vector Elements* node to combine a group of paths into one path (composed of subpaths with the original geometry).

> Also, now there's 2 strokes in line with eachother (path -> transform -> stroke -> layer(/group?) -> stroke - I guess the later stroke overrides the earlier?

Yes, at least right now, vector data can have only one of each appearance property (fill, stroke, opacity, and blend mode). Later appearance-setting nodes replace the previously set appearance. To remove an appearance, you can set its color to none (the red slash choice in the color picker).

> Also, I didn't see a group option anywhere.

The button does exist at the top of the Layers panel, it's the folder icon which says "Group Selected (Ctrl G)" in its tooltip. But you're right, that's an oversight that it's missing from *Layer* > *Group Selected*. Thanks for pointing that out, I'll add it. Besides that button in the Layers panel, layers can also be grouped by dragging them into another blank layer, which can be created via the `[+]` button next to the Group Selected button. That makes me realize we should also have a *Layer* > *New* menu bar option, which I'll add now as well.

**Reply by andrewbaxter** [reply]
> Later appearance-setting nodes replace the previously set appearance

Ah, okay! So the `stroke` node doesn't modify/change geometry.  I was thinking it was like "Stroke to path" in inkscape, but it just sets existing properties.  Thanks! I think I'm all clear now.

> it's the folder icon

Ah, my bad, you're right! I saw that icon and thought it was something else I think.

Anyways, thanks for all the answers again! This is a really cool project, and both the manual path editing and ui responsiveness are great!

**Reply by Keavon** [reply]
> Ah, okay! So the `stroke` node doesn't modify/change geometry. I was thinking it was like "Stroke to path" in inkscape, but it just sets existing properties. Thanks! I think I'm all clear now.

We have the *Solidify Stroke* node for that.
